[15.0][IMP] dms_field: Add support to drag & drop

CarlosRoca13 opened the pull request:
cc @Tecnativa TT48179

ping @victoralmau @chienandalu 

With this changes we add the drag and drop option for dms fields. On next gif you can see an example:
![Ejemplo dms_field](https://github.com/OCA/dms/assets/35952655/d4217830-efd0-4586-909a-3e58b128daf0)

CarlosRoca13 committed:
[IMP] dms_field: Add support to drag & drop

chienandalu commented on dms_field/static/src/js/views/many_drop_target.js:
```diff
@@ -0,0 +1,69 @@
+/* global base64js*/
+/* Copyright 2020 Creu Blanca
+ * Copyright 2021 Tecnativa - Alexandre D. Díaz
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("dms_field.DragDrop", function (require) {
```
Is this copied from somewhere else?

CarlosRoca13 commented:
Changed

pedrobaeza commented:
Does this need to depend on `web_drop_target`?

CarlosRoca13 commented:
dms is depending yet on that module

legalsylvain reviewed (approved):
Hi @CarlosRoca13. Nice work !

tested on runboat. works like a charm.

pedrobaeza commented:
Isn't possible to preserve previous folding state in the tree when dragging the file? Maybe adding the element by hand with jstree API instead of refreshing whole one.

CarlosRoca13 commented:
What do you mean? When I drag the file I don't see the directories folding...
![keepling_state](https://github.com/OCA/dms/assets/35952655/fba4d413-c9fb-4ba1-9e6e-239201adcb0a)

pedrobaeza commented:
The problem is reproducible when you drop on a folder with subfolders that are folded:

![Peek 11-03-2024 08-36](https://github.com/OCA/dms/assets/7165771/844e3b84-7aa4-4ac2-ad2d-c9f2c1602e76)

CarlosRoca13 committed:
[IMP] dms_field: Open all directories at first charge but not set open as default state

Before this change, when lauching the reload, the child directories
of the reloaded one are opened too.

By doing at this step and not setting the default state, the reloads are
done just for the implied nodes.

Furthermore, the code is improved to just reload the directories that
have being changed.

CarlosRoca13 commented:
@pedrobaeza Solved and improved the code to avoid full recharging when adding new directory or new field using the wizard :smile:

pedrobaeza commented:
Still incorrect. Now it folds all the subfolders:

![Peek 11-03-2024 13-01](https://github.com/OCA/dms/assets/7165771/a69dc029-dc5d-4a28-972d-c577b99f39ae)

CarlosRoca13 commented:
It should be working correctly right now :)

pedrobaeza reviewed (approved):
Great, thanks.

/ocabot merge minor
